pennylane_lightning/src/util/Memory.hpp

```diff
@@ -80,7 +80,16 @@ template <class T> class AlignedAllocator {
     const uint32_t alignment_;
 
   public:
+    using pointer = T *;
+    using const_pointer = const T *;
+    using void_pointer = void *;
+    using const_void_pointer = const void *;
     using value_type = T;
+    using size_type = std::size_t;
+    using difference_type = std::ptrdiff_t;
+    using propagate_on_container_move_assignment = std::true_type;
+
+    template <typename U> struct rebind { using other = AlignedAllocator<U>; };
```

Comment:
Same for rebind. Deprecated in C++17 and removed in C++20.

Reply:
As above.